docs/support/upgrade.md

```diff
@@ -0,0 +1,223 @@
+## Upgrading your project
+
+In this section we will discuss changes and upgrade advice for users upgrading from versions of Nitric reliant on a nitric.yaml file.
+
+### What you’ll learn
+
+- Install the new Nitric CLI
+- Upgrade Nitric projects targetting the X CLI
+
+> &#x26a0;&#xfe0f; ** Info ** - Ensure all source code is safely backed up in repository.
+
+### What's changed?
+
+The Node SDK now utilizes a 'Config-as-Code' design pattern.
+
+- Resources, permissions and configuration will infer automatically from your code.
+
+The API defintions have minor changes to support the new pattern.
+
+- Examples on how to use the API can be found in the [documentation](/docs).
+
+### Why the change?
+
+Config requires alot of research, Nitric eliminates 1000s of lines of config that you'll need to manage.
```

Comment:
This line reads a bit strangely after the previous paragraph. I feel like elaborating more on how config as code enables this.

e.g. instead of `Config requires alot of research` (have to read lots of docs to understand) something like `Config as code is expressive and self-documentating` (saves time by not having to read documentation).

Probably better ways to write that, that was just my first impression reading it.

Reply:
Good suggestion, will redraft this line.